src/USER-ADIOS/dump_custom_adios.h

```diff
@@ -0,0 +1,94 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifdef DUMP_CLASS
+
+DumpStyle(custom/adios,DumpCustomADIOS)
+
+#else
+
+#ifndef LMP_DUMP_CUSTOM_ADIOS_H
+#define LMP_DUMP_CUSTOM_ADIOS_H
+
+#include "dump_custom.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include "adios2.h"
+
+namespace LAMMPS_NS {
+
+class DumpCustomADIOS : public DumpCustom {
+ public:
+  DumpCustomADIOS(class LAMMPS *, int, char **);
+  virtual ~DumpCustomADIOS();
+
+ protected:
+
+  const std::string ioName="custom";   // name of adios group, referrable in adios2_config.xml
+  adios2::ADIOS *ad = nullptr; // adios object
+  adios2::IO io;    // adios group of variables and attributes in this dump
+  adios2::Engine fh; // adios file/stream handle object
+  adios2::Variable<double> varAtoms; // one ADIOS output variable we need to change
+  uint64_t groupsize; // pre-calculate # of bytes written per processor in a step before writing anything
+  std::vector<std::string>columnNames; // list of column names for the atom table (individual list of 'columns' string)
```

Comment:
please don't use STL container classes in "style" headers / style classes. See my previous comments about reducing issues from including all "style" headers at the same time in some files.